pkg/dotnet/core_deps/parse_test.go

```diff
@@ -15,19 +15,251 @@ import (
 
 func TestParse(t *testing.T) {
 	vectors := []struct {
-		file    string // Test input file
-		want    []types.Library
-		wantErr string
+		file     string // Test input file
+		wantLibs []types.Library
+		wantDeps []types.Dependency
+		wantErr  string
 	}{
+		{
+			file: "testdata/MyExample.deps.json",
+			wantLibs: []types.Library{
+				{ID: "MyWebApp@1.0.0", Name: "MyWebApp", Version: "1.0.0", Locations: []types.Location{{StartLine: 148, EndLine: 152}}},
```

Comment:
Is there reason to add `project` to result?
We didn't include this earlier: https://github.com/aquasecurity/go-dep-parser/blob/6eff4752bcbe21782f5521ad234b0ea501ad8410/pkg/dotnet/core_deps/parse.go#L35

Reply:
yes, because otherwise you can't extract a full and complete dependency tree if you need to parse the SBOM

Reply:
This makes sense for SBOM, but can be confusing when used by default (like` json` format).
We save project in dependencies. This can be confusing.

Perhaps we need to add new type or field for `Library` for project name/version. But I think it is change for another PR.

We use `application` componentType for SBOM - https://github.com/aquasecurity/go-dep-parser/pull/258#discussion_r1353970059
Previously this was enough.